crates/http/src/http.rs

```diff
@@ -27,29 +27,42 @@ fn http_proxy_from_env() -> Option<isahc::http::Uri> {
         };
     }
 
-    try_env!(
-        "ALL_PROXY",
-        "all_proxy",
-        "HTTPS_PROXY",
-        "https_proxy",
-        "HTTP_PROXY",
-        "http_proxy"
-    );
-    None
+    proxy
+        .and_then(|input| {
+            input
+                .parse::<isahc::http::Uri>()
+                .inspect_err(|e| log::error!("Error parsing proxy settings: {}", e))
+                .ok()
+        })
+        .or_else(|| {
+            try_env!(
+                "ALL_PROXY",
+                "all_proxy",
+                "HTTPS_PROXY",
+                "https_proxy",
+                "HTTP_PROXY",
+                "http_proxy"
+            );
+            None
+        })
 }
 
 /// An [`HttpClient`] that has a base URL.
 pub struct HttpClientWithUrl {
     base_url: Mutex<String>,
     client: Arc<dyn HttpClient>,
+    proxy_settings: Option<String>,
```

Comment:
How about this named `proxy`, like `proxy` method in isahc::HttpClient.

Reply:
Makes sense.